elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java

```diff
@@ -536,6 +565,58 @@ private static <T> Predicate<T> hasMember(Path fieldPath, List<Object> values, R
         };
     }
 
+    private static <T> Predicate<T> subsetOf(Path fieldPath, List<Object> values, RequestScope requestScope) {
+        return (T entity) -> {
+            Type<?> valueClass = fieldPath.lastElement().get().getFieldType();
+
+            Object leftHandSide = getFieldValue(entity, fieldPath, requestScope);
+
+            BiPredicate<Object, Object> predicate = (a, b) -> a.equals(b);
+
+            List<?> rightHandSide = values.stream().map(value -> CoerceUtil.coerce(value, valueClass)).toList();
+
+            if (leftHandSide instanceof Collection<?> collection && !valueClass.isAssignableFrom(COLLECTION_TYPE)) {
+                for (Object left : collection) {
+                    if (!rightHandSide.stream().anyMatch(object -> predicate.test(left, object))) {
+                        return false;
+                    }
+                }
+            } else {
+                if (leftHandSide != null
+                        && !rightHandSide.stream().anyMatch(object -> predicate.test(leftHandSide, object))) {
+                    return false;
+                }
+            }
+            return true;
+        };
+    }
+
+    private static <T> Predicate<T> supersetOf(Path fieldPath, List<Object> values, RequestScope requestScope) {
+        return (T entity) -> {
+            Type<?> valueClass = fieldPath.lastElement().get().getFieldType();
+
+            Object leftHandSide = getFieldValue(entity, fieldPath, requestScope);
+
+            BiPredicate<Object, Object> predicate = (a, b) -> a.equals(b);
+
+            List<?> rightHandSide = values.stream().map(value -> CoerceUtil.coerce(value, valueClass)).toList();
+
+            if (leftHandSide instanceof Collection<?> collection && !valueClass.isAssignableFrom(COLLECTION_TYPE)) {
+                for (Object right : rightHandSide) {
+                    if (!collection.stream().anyMatch(object -> predicate.test(right, object))) {
+                        return false;
+                    }
+                }
+            } else {
+                for (Object right : rightHandSide) {
+                    if (!predicate.test(right, leftHandSide)) {
```

Comment:
What about the case where both left hand side is a collection and the valueClass is a collection.  The predicate here is testing equality of the collection in that case with a member on the right hand side.

Reply:
After looking at it again, I'm actually not sure if the `valueClass` can ever be assignable from a collection type.

In the `testSubsetOfOnAttributes` test in `com.yahoo.elide.tests.FilterIT` filtering on book.awards the `valueClass` is a string type.

```java
public class Book extends BaseId {
    ...
    private Collection<String> awards = new ArrayList<>();
}
```

For it to be a collection type, it would look something like
```java
public class Book extends BaseId {
    ...
    private Collection<Collection<String>> collection = new ArrayList<>();
}
```

I guess the logic is a little misleading as the else condition only happens when `leftHandSide` is null as `leftHandSide instanceof null` is false. I shall fix the logic to make it clearer.